### PR TITLE
Optimize ci testcase  for kubeedge

### PR DESCRIPTION
**Pull request by kadisi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Optimize ci testcase  for kubeedge

1 bugfix:local tests generate edge.db in source code
2 bugfix: local test will modify the configuration file in the source code
3 Optimize testcase  for kubeedge，Delete useless testcase
```

### CODE REVIEW

File: edge/pkg/edgehub/controller_test.go [hunk 1]
```diff
@@ -182,72 +80,20 @@ func TestNewEdgeHubController(t *testing.T) {
...
-		t.Run(tt.name, func(t *testing.T) {
-			edgeHubConfig := config.GetConfig()
-			edgeHubConfig.WSConfig = tt.webSocketConfig
-			edgeHubConfig.CtrConfig = tt.controllerConfig
-			if err := tt.controller.initial(tt.ctx); err != tt.expectedError {
-				t.Errorf("EdgeHubController_initial() error = %v, expectedError %v", err, tt.expectedError)
-			}
-		})
-	}
-}
```
Comment (fisherxu) [inline edge/pkg/edgehub/controller_test.go]: Keep this case not removed?

File: edge/pkg/edgehub/controller_test.go [hunk 2]
```diff
@@ -588,48 +494,3 @@ func TestKeepalive(t *testing.T) {
...
-				msg, err := testContext.Receive(module.TwinGroup)
-				if err != nil {
-					t.Errorf("Error in receiving message from twin group: %v", err)
-				} else if msg.Content != tt.content {
-					t.Errorf("TestPubConnectInfo() Content of message received in twin group : %v, want: %v", msg.Content, tt.content)
-				}
-			})
-		})
-	}
-}
```
Comment (fisherxu) [inline edge/pkg/edgehub/controller_test.go]: Not removed?

File: edge/pkg/devicetwin/dtclient/sqlite_test.go [hunk 3]
```diff
@@ -291,7 +356,6 @@ func TestQueryTwinAll(t *testing.T) {
 	for _, test := range cases {
 		t.Run(test.name, func(t *testing.T) {
 			querySeterMock.EXPECT().All(gomock.Any()).SetArg(0, *fakeTwin).Return(test.allReturnInt, test.allReturnErr).Times(1)
-			querySeterMock.EXPECT().Filter(gomock.Any(), gomock.Any()).Return(test.filterReturn).Times(1)
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtclient/sqlite_test.go]: Why remove this line?

File: edge/pkg/devicetwin/dtmanager/device_test.go [hunk 4]
```diff
@@ -363,18 +432,18 @@ func TestDeviceUpdated(t *testing.T) {
 			commitTimes:      1,
 			beginTimes:       1,
 			filterReturn:     querySeterMock,
-			filterTimes:      6,
+			filterTimes:      0,
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/device_test.go]: Why set to 0？

File: edge/pkg/devicetwin/dtmanager/device_test.go [hunk 5]
```diff
@@ -363,18 +432,18 @@ func TestDeviceUpdated(t *testing.T) {
...
 			insertReturnInt:  int64(1),
 			insertReturnErr:  nil,
 			insertTimes:      1,
 			deleteReturnInt:  int64(1),
 			deleteReturnErr:  nil,
-			deleteTimes:      1,
+			deleteTimes:      0,
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/device_test.go]: ditto

File: edge/pkg/devicetwin/dtmanager/device_test.go [hunk 6]
```diff
@@ -363,18 +432,18 @@ func TestDeviceUpdated(t *testing.T) {
...
 			updateReturnInt:  int64(1),
 			updateReturnErr:  nil,
-			updateTimes:      1,
+			updateTimes:      0,
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/device_test.go]: ditto？

File: edge/pkg/devicetwin/dtmanager/device_test.go [hunk 7]
```diff
@@ -363,18 +432,18 @@ func TestDeviceUpdated(t *testing.T) {
...
 			queryTableReturn: querySeterMock,
-			queryTableTimes:  2,
+			queryTableTimes:  0,
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/device_test.go]: ditto

File: edge/pkg/devicetwin/dtmanager/twin_test.go [hunk 8]
```diff
@@ -459,19 +482,27 @@ func TestDealDeviceTwin(t *testing.T) {
...
-			mockQuerySeter.EXPECT().Filter(gomock.Any(), gomock.Any()).Return(test.filterReturn).Times(0)
-			mockOrmer.EXPECT().Insert(gomock.Any()).Return(test.allReturnInt, test.allReturnErr).Times(1)
-			mockQuerySeter.EXPECT().Delete().Return(test.allReturnInt, test.allReturnErr).Times(0)
-			mockQuerySeter.EXPECT().Update(gomock.Any()).Return(test.allReturnInt, test.allReturnErr).Times(1)
-			mockOrmer.EXPECT().QueryTable(gomock.Any()).Return(test.queryTableReturn).Times(0)
+			mockOrmer.EXPECT().Rollback().Return(nil).Times(test.rollbackNums)
+			mockOrmer.EXPECT().Begin().Return(nil).MaxTimes(test.beginNums)
+			mockOrmer.EXPECT().Commit().Return(nil).MaxTimes(test.commitNums)
+			mockOrmer.EXPECT().Insert(gomock.Any()).Return(test.allReturnInt, test.allReturnErr).MaxTimes(test.insertNums)
+			mockQuerySeter.EXPECT().Filter(gomock.Any(), gomock.Any()).Return(test.filterReturn).Times(test.filterNums)
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/twin_test.go]: why remove these lines?

File: edge/pkg/devicetwin/dtmanager/membership_test.go [hunk 9]
```diff
@@ -189,20 +174,39 @@ func TestDealMembershipUpdatedInvalidContent(t *testing.T) {
...
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	ormerMock = beego.NewMockOrmer(mockCtrl)
+	querySeterMock = beego.NewMockQuerySeter(mockCtrl)
+	dbm.DBAccess = ormerMock
+
+	ormerMock.EXPECT().Begin().Return(nil)
+	ormerMock.EXPECT().Insert(gomock.Any()).Return(int64(1), nil).Times(1)
+	ormerMock.EXPECT().Commit().Return(nil)
+	querySeterMock.EXPECT().Filter(gomock.Any(), gomock.Any()).Return(querySeterMock).Times(0)
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership_test.go]: Times(0)?

File: edge/pkg/devicetwin/dtmanager/membership_test.go [hunk 10]
```diff
@@ -309,55 +353,3 @@ func TestDealGetMembershipInValid(t *testing.T) {
...
-		ID:    "DeviceA",
-		Name:  "Router",
-		State: "unknown",
-	}}
-	var b = dttype.BaseMessage{
-		EventID: "eventid",
-	}
-
-	Removed(dtc, d, b, true)
-}*/
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership_test.go]: keep them here?

File: edge/Makefile [hunk 11]
```diff
@@ -19,7 +19,7 @@ endif
 
 export GOARCHAIUS_CONFIG_PATH=$(CURDIR)
 test:
-	find ${TEST_DIR} -name "*_test.go"| xargs -I{} dirname {} | uniq | xargs -I{} go test ${T} {}
+	find ${TEST_DIR} -name "*_test.go"| xargs -I{} dirname {} | uniq | xargs -I{} go test -v ${T} {}
```
[the remark this answers is not in the document]
Comment (kadisi) [reply edge/Makefile]: ok

File: edge/pkg/devicetwin/dtclient/deviceattr_db_test.go [hunk 12]
```diff
@@ -478,11 +564,12 @@ func TestDeviceAttrTrans(t *testing.T) {
...
 			ormerMock.EXPECT().Commit().Return(nil).Times(test.commitTimes)
 			ormerMock.EXPECT().Begin().Return(nil).Times(test.beginTimes)
-			querySeterMock.EXPECT().Filter(gomock.Any(), gomock.Any()).Return(test.filterReturn).Times(test.filterTimes)
 			ormerMock.EXPECT().Insert(gomock.Any()).Return(test.insertReturnInt, test.insertReturnErr).Times(test.insertTimes)
+			ormerMock.EXPECT().QueryTable(gomock.Any()).Return(test.queryTableReturn).Times(test.queryTableTimes)
+
+			querySeterMock.EXPECT().Filter(gomock.Any(), gomock.Any()).Return(test.filterReturn).Times(test.filterTimes)
 			querySeterMock.EXPECT().Delete().Return(test.deleteReturnInt, test.deleteReturnErr).Times(test.deleteTimes)
 			querySeterMock.EXPECT().Update(gomock.Any()).Return(test.updateReturnInt, test.updateReturnErr).Times(test.updateTimes)
-			ormerMock.EXPECT().QueryTable(gomock.Any()).Return(test.queryTableReturn).Times(test.queryTableTimes)
```
[the remark this answers is not in the document]
Comment (kadisi) [reply edge/pkg/devicetwin/dtclient/deviceattr_db_test.go]: not move , this lines,  has Move it to the top few lines.

File: edge/pkg/devicetwin/dtclient/sqlite_test.go (hunk 3 above)
Comment (kadisi) [reply edge/pkg/devicetwin/dtclient/sqlite_test.go]: QueryTwinAll() function did not call the relevant Filter() function.  no need mock

File: edge/pkg/devicetwin/dtmanager/membership_test.go (hunk 9 above)
Comment (fisherxu) [reply edge/pkg/devicetwin/dtmanager/membership_test.go]: Just delete this line.

File: edge/pkg/devicetwin/dtmanager/membership_test.go [hunk 13]
```diff
@@ -213,19 +217,39 @@ func TestDealMembershipUpdatedValidAddedDevice(t *testing.T) {
...
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	ormerMock = beego.NewMockOrmer(mockCtrl)
+	querySeterMock = beego.NewMockQuerySeter(mockCtrl)
+	dbm.DBAccess = ormerMock
+
+	ormerMock.EXPECT().Begin().Return(nil).Times(0)
+	ormerMock.EXPECT().Insert(gomock.Any()).Return(int64(1), nil).Times(0)
+	ormerMock.EXPECT().Commit().Return(nil).Times(0)
+	ormerMock.EXPECT().QueryTable(gomock.Any()).Return(querySeterMock).Times(0)
```
Comment (fisherxu) [inline edge/pkg/devicetwin/dtmanager/membership_test.go]: delete this line.

Comment (kadisi) [reply edge/pkg/devicetwin/dtmanager/membership_test.go]: ok

**Comment by fisherxu** [comment]
/lgtm
Thanks!

**Pull request merged**